### PR TITLE
Add the ability to add non-breaking text to a TextFlow

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

**Comment by Tom94** [comment]
I think a better implementation would be to make use of the non-breaking space character. In `AddTextNonBreaking` simply call `AddText` where all spaces in the original text have been replaced by a non-breaking space. The regex will then naturally not split across these.

**Comment by peppy** [comment]
The use case here is periods, not spaces

**Comment by Tom94** [comment]
Wait, words are split by periods via the regex? What?!? Periods should only implicitly result in breaks because of the breaking-space behind them. "Bla.Bla" should _never_ result in a linebreak,

**Comment by peppy** [comment]
Sounds correct, the regex should be changed.

**Comment by Tom94** [comment]
Probably best to not use a regex at all and instead just split via `Char.IsSeparator`. Important to keep non-`Char.IsWhitespace` separators in the first word of each split, though.

**Comment by peppy** [comment]
Rewritten

File: osu.Framework/Graphics/Containers/TextFlowContainer.cs [hunk 1]
```diff
@@ -200,6 +200,33 @@ private IEnumerable<SpriteText> addLine(TextLine line, bool newLineIsParagraph)
             return sprites;
         }
 
+        private string[] splitWords(string text)
+        {
+            var words = new List<string>();
+            var builder = new StringBuilder();
+
+            bool pendingNewWord = false;
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/TextFlowContainer.cs]: Just make it a `for` loop and check `i == 0 || IsSeparator(text[i-1]) || IsControl(text[i-1])`. Gets rid of the local state `pendingNewWord`.

Comment (peppy) [inline osu.Framework/Graphics/Containers/TextFlowContainer.cs]: Doesn't that change the behaviour of consecutive separator characters?

Comment (Tom94) [inline osu.Framework/Graphics/Containers/TextFlowContainer.cs]: Yeah, but I think it's to the better. We don't want consecutive separator chars to be put in the same word which is non-breaking. We want to allow breaks after _every_ separator char, so imho we should put each separator char into an individual word.

**Pull request merged**